Makefile

```diff
@@ -143,25 +113,65 @@ docs-%:
 
 ######################################################
 #
-# CI targets
+# Wasm targets
 #
 ######################################################
 
-.PHONY: travis-build
-travis-build: wasm-build
-	@# this image is used in `Dockerfile` for image-quick
-	$(DOCKER) build -t build-$(BUILD_COMMIT) --build-arg GOVERSION=$(GOVERSION) -f Dockerfile.build .
-	@# the '/.' means "don't create the directory, copy its content only"
-	@# these are copied our to be used from the s3 upload targets
-	@# note: we don't bother cleaning up the container created here
-	$(DOCKER) cp "$$($(DOCKER) create build-$(BUILD_COMMIT)):/out/." .
+.PHONY: wasm-lib-build
+wasm-lib-build:
+ifeq ($(DOCKER_INSTALLED), 1)
+	@$(MAKE) -C wasm build
+	cp wasm/_obj/opa.wasm internal/compiler/wasm/opa/opa.wasm
+else
+	@echo "Docker not installed. Skipping OPA-WASM library build."
+endif
+
+.PHONY: wasm-lib-test
+wasm-lib-test:
+ifeq ($(DOCKER_INSTALLED), 1)
+	@$(MAKE) -C wasm test
+else
+	@echo "Docker not installed. Skipping OPA-WASM library test."
+endif
+
+.PHONY: wasm-lib-clean
+wasm-lib-clean:
+	@$(MAKE) -C wasm clean
+
+.PHONY: wasm-rego-test
+wasm-rego-test: generate
+ifeq ($(DOCKER_INSTALLED), 1)
+	GOVERSION=$(GOVERSION) ./build/run-wasm-rego-tests.sh
+else
+	@echo "Docker not installed. Skipping Rego-WASM test."
+endif
+
+.PHONY: wasm-rego-testgen-install
+wasm-rego-testgen-install:
+	$(GO) install -i ./test/wasm/cmd/wasm-rego-testgen
+
+.PHONY: wasm-test
+wasm-test: wasm-lib-test wasm-rego-test
+
+######################################################
+#
+# CI targets
+#
+######################################################
 
-.PHONY: travis-test
-travis-test: travis-build wasm-test wasm-rego-test
-	$(DOCKER) run build-$(BUILD_COMMIT) make go-test perf check
+.PHONY: travis
+travis:
+	$(DOCKER) run \
+		-u $(shell id -u):$(shell id -g) \
+		--rm \
+		-v $(PWD):/go/src/github.com/open-policy-agent/opa \
+		-w /go/src/github.com/open-policy-agent/opa \
```

Comment:
If we wanted to lean in on the module stuff we could put this at just like `/opa` and move off the GOSRC path 😄

Reply:
That is going to be a hard one to unlearn. I'll leave as-is for now since I don't have a compelling alternative.

Reply:
I've leaned in to using /src in the container.